File: api/CarDecoration.API/CarDecoration.API/Services/UserService.cs
using CarDecoration.API.Data;
using CarDecoration.API.DTOs;
using CarDecoration.API.Helpers;
using CarDecoration.API.Models;
using Microsoft.EntityFrameworkCore;

namespace CarDecoration.API.Services;

public class UserService
{
    private readonly AppDbContext _db;
    private readonly ICurrentUserService _currentUser;

    public UserService(AppDbContext db, ICurrentUserService currentUser)
    {
        _db = db;
        _currentUser = currentUser;
    }

    public async Task<UserProfileResponse> GetProfileAsync()
    {
        var userId = _currentUser.UserId
            ?? throw new Exception("المستخدم غير موجود");

        var user = await _db.Users
            .AsNoTracking()
            .FirstOrDefaultAsync(u => u.Id == userId)
            ?? throw new Exception("المستخدم غير موجود");

        var vehicleCount = await _db.Vehicles
            .CountAsync(v => v.OwnerId == userId);

        var activeStatuses = new[] { RequestStatus.Open, RequestStatus.ShopSelected, RequestStatus.InProgress };
        var activeRequestCount = await _db.Requests
            .CountAsync(r => r.CustomerId == userId && activeStatuses.Contains(r.Status));

        var reviewCount = await _db.Reviews
            .CountAsync(r => r.Request.CustomerId == userId);

        return new UserProfileResponse(
            user.FullName,
            user.Phone,
            user.Email,
            vehicleCount,
            activeRequestCount,
            reviewCount
        );
    }

    public async Task<UserProfileResponse> UpdateProfileAsync(UpdateProfileRequest req)
    {
        var userId = _currentUser.UserId
            ?? throw new Exception("المستخدم غير موجود");

        var user = await _db.Users.FirstOrDefaultAsync(u => u.Id == userId)
            ?? throw new Exception("المستخدم غير موجود");

        var trimmedName = req.FullName?.Trim() ?? "";
        var trimmedPhone = req.Phone?.Trim() ?? "";

        if (trimmedName.Length < 2)
            throw new Exception("الاسم يجب أن يكون حرفين على الأقل");

        if (!string.IsNullOrEmpty(trimmedPhone) && trimmedPhone != user.Phone)
        {
            if (await _db.Users.AnyAsync(u => u.Phone == trimmedPhone && u.Id != userId))
                throw new Exception("رقم الجوال مستخدم من حساب آخر");
            user.Phone = trimmedPhone;
        }

        user.FullName = trimmedName;
        await _db.SaveChangesAsync();

        return await GetProfileAsync();
    }
}

File: car_decoration_app/lib/services/user_service.dart
import 'api_client.dart';

class UserProfileData {
  final String fullName;
  final String phone;
  final String email;
  final int vehicleCount;
  final int activeRequestCount;
  final int reviewCount;

  const UserProfileData({
    required this.fullName,
    required this.phone,
    required this.email,
    required this.vehicleCount,
    required this.activeRequestCount,
    required this.reviewCount,
  });

  factory UserProfileData.fromJson(Map<String, dynamic> json) => UserProfileData(
    fullName: json['fullName'] as String? ?? '',
    phone: json['phone'] as String? ?? '',
    email: json['email'] as String? ?? '',
    vehicleCount: json['vehicleCount'] as int? ?? 0,
    activeRequestCount: json['activeRequestCount'] as int? ?? 0,
    reviewCount: json['reviewCount'] as int? ?? 0,
  );
}

class UserService {
  static Future<UserProfileData> getProfile() async {
    final res = await ApiClient.dio.get('/api/users/profile');
    return UserProfileData.fromJson(res.data as Map<String, dynamic>);
  }

  static Future<UserProfileData> updateProfile({
    required String fullName,
    required String phone,
  }) async {
    final res = await ApiClient.dio.put('/api/users/profile', data: {
      'fullName': fullName,
      'phone': phone,
    });
    final data = UserProfileData.fromJson(res.data as Map<String, dynamic>);
    await ApiClient.updateCachedProfile(fullName: data.fullName, phone: data.phone);
    return data;
  }
}
